Add (de)serialization for Property/Records

M3rs opened the pull request:
Properties have a backing field named like '<MyField>k__BackingField'
which can get used for properties and record types.

i.e.

```
public class MyClass
{
	public string MyString { get; set; }
}
```

When looking at `typeof(MyClass)`, there will be a private field
`private string <MyString>k__BackingField`.

- Add to `GetCompositeDeserializer` and `GetCompositeSerializer` the
  ability to read properties.
- Add to tests examples of class with Properties and record type

M3rs committed:
Add (de)serialization for Property/Records

Properties have a backing field named like '<MyField>k__BackingField'
which can get used for properties and record types.

i.e.

```
public class MyClass
{
	public string MyString { get; set; }
}
```

When looking at `typeof(MyClass)`, there will be a private field
`private string <MyString>k__BackingField`.

- Add to `GetCompositeDeserializer` and `GetCompositeSerializer` the
  ability to read properties.

SamboyCoding reviewed (changes requested):
Please gitignore your .vs folder - test logs and user-specific configuration should not be pushed.

SamboyCoding commented on Tomlet/TomlSerializationMethods.cs:
```diff
@@ -136,7 +136,7 @@ internal static Deserialize<object> GetCompositeDeserializer(Type type)
             {
                 //Get all instance fields
-                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.SetProperty);
```
I don't think this change is required, `SetProperty` only applies when calling methods, from what I can see on the docs, and regardless your change isn't required by the code you wrote below.

SamboyCoding commented on Tomlet/TomlSerializationMethods.cs:
```diff
@@ -222,7 +227,7 @@ internal static Serialize<object> GetCompositeSerializer(Type type)
 
                 //Get all instance fields
-                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+                var fields = type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.GetProperty);
```
Same as above, I don't think `GetProperty` needs to be specified here.

SamboyCoding commented:
Other than those minor details, nice PR! Very appreciated, and thanks for writing additional tests too.

M3rs committed:
Clean up per feedback

- ignore .vs/ and clean up repo
- remote Get/SetProperty BindingFlags (not needed)

M3rs commented:
Ah, good call out on the BindingFlags. Cleaned up that and the gitignore.

SamboyCoding commented:
Alright, merging this now, will try and get a release out tonight.

SamboyCoding commented:
As a slight aside, your local git email doesn't match your github one, so these commits aren't attributed to you.

M3rs commented:
Ah, I'll get that sorted, thanks

SamboyCoding commented:
This is now live in 2.0.0. Sorry for the delay in getting this out in a release!
